src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs

```diff
@@ -4,55 +4,30 @@
-            => _globalOptionService = globalOptionService;
-
-        public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
-            => new OptionService(_globalOptionService);
-
-        internal sealed class OptionService : IOptionService
-        {
-            private readonly IGlobalOptionService _globalOptionService;
-
-            public OptionService(IGlobalOptionService globalOptionService)
-            {
-                _globalOptionService = globalOptionService;
-            }
-
-            // Simple forwarding functions.
-            public SolutionOptionSet GetOptions() => _globalOptionService.GetOptions(this);
-            public object? GetOption(OptionKey optionKey) => _globalOptionService.GetOption(optionKey);
-            public T? GetOption<T>(Option<T> option) => _globalOptionService.GetOption(option);
-            public T? GetOption<T>(Option2<T> option) => _globalOptionService.GetOption(option);
-            public T? GetOption<T>(PerLanguageOption<T> option, string? languageName) => _globalOptionService.GetOption(option, languageName);
-            public T? GetOption<T>(PerLanguageOption2<T> option, string? languageName) => _globalOptionService.GetOption(option, languageName);
-            public IEnumerable<IOption> GetRegisteredOptions() => _globalOptionService.GetRegisteredOptions();
-            public bool TryMapEditorConfigKeyToOption(string key, string? language, [NotNullWhen(true)] out IEditorConfigStorageLocation2? storageLocation, out OptionKey optionKey) => _globalOptionService.TryMapEditorConfigKeyToOption(key, language, out storageLocation, out optionKey);
-            public ImmutableHashSet<IOption> GetRegisteredSerializableOptions(ImmutableHashSet<string> languages) => _globalOptionService.GetRegisteredSerializableOptions(languages);
-            public void SetOptions(OptionSet optionSet) => _globalOptionService.SetOptions(optionSet);
-            public void RegisterWorkspace(Workspace workspace) => _globalOptionService.RegisterWorkspace(workspace);
-            public void UnregisterWorkspace(Workspace workspace) => _globalOptionService.UnregisterWorkspace(workspace);
-        }
-    }
+    public IGlobalOptionService GlobalOptions { get; }
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public LegacyWorkspaceOptionService(IGlobalOptionService globalOptionService)
+        => GlobalOptions = globalOptionService;
+
+    public void RegisterWorkspace(Workspace workspace)
+        => GlobalOptions.RegisterWorkspace(workspace);
+
+    public void UnregisterWorkspace(Workspace workspace)
+        => GlobalOptions.UnregisterWorkspace(workspace);
+
+    public object? GetOption(OptionKey key)
+        => GlobalOptions.GetOption(key);
+
+    public void SetOptions(OptionSet optionSet, IEnumerable<OptionKey> optionKeys)
+        => GlobalOptions.SetOptions(optionSet, optionKeys);
```

Comment:
if this all sits on IGlobalOptionService then can we have consumers of ILegacyWorkspaceOptionService just use IGlobalOptionService?

(i'm a little confused as to what is being accomplished here).

Reply:
`IGlobalOptionService` is not a workspace service it's MEF service. We need a workspace service that we can access from Workspace constructor.